collectors/30/kafka_offset_collector.py

```diff
@@ -40,60 +45,78 @@ def format_kafka_consumer_offset_tsd_key(consumer_group, topic, partition, offse
 # KAFKA OFFSET MONITORING
 
 
-def get_partitions(zk, group, topic):
-    return zk.get_children(KafkaPaths.topic_offsets(group, topic))
+def get_partitions(zk, topic):
+    path = KafkaPaths.partitions(topic)
+    return map(int, zk.get_children(path))
+
+
+def get_kafka_topics(zk):
+    path = KafkaPaths.topics()
+    return [str(topic) for topic in zk.get_children(path) if topic not in BLACKLIST]
 
 
 def get_consumer_group_offset(zk, group, topic, partition):
     path = KafkaPaths.consumer_topic_partition(group, topic, partition)
-    return int(zk.get(path)[0])
-
+    try:
+        return int(zk.get(path)[0])
+    except NoNodeError:
+        return 0
 
-def report_broker_info(kafka, zk, topic):
-    def get_partitions(topic):
-        path = KafkaPaths.broker_partitions(topic)
-        return map(int, zk.get_children(path))
 
+def report_broker_info(env, kafka, zk, topic):
     total_offset = 0
-    for partition in get_partitions(topic):
+    for partition in get_partitions(zk, topic):
         offset_response = kafka.send_offset_request([OffsetRequest(topic, partition, -1, 1)])[0]
         offset = offset_response.offsets[0]
         total_offset += offset
-        print format_kafka_broker_offset_tsd_key(topic, partition, offset)
+        print format_kafka_broker_offset_tsd_key(env, topic, partition, offset)
     return total_offset
 
 
-def report():
-    # see the monitoring.json chef role
-    zk_quorums = os.getenv('MONITORED_ZOOKEEPER_QUORUMS')
-    kafka_brokers = os.getenv('KAFKA_BROKERS')
+def add_dns_domain(hosts):
+    def add_dns_domain_to_host(host):
+        port = None
+        if ':' in host:
+            host, port = host.split(':', 1)
```

Comment:
For a string like `ahost:80808:someotherstuff` this will parse `80808:someotherstuff` as `port`, is that the intended behavior? Or do you want something like 
`host.split(':')[:2]` since you're only interested in the first 2 items?